include/exadg/postprocessor/time_control.cpp

```diff
@@ -0,0 +1,127 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+// C/C++
+#include <limits>
+
+// deal.II
+#include <deal.II/base/exceptions.h>
+
+// ExaDG
+#include <exadg/postprocessor/time_control.h>
+
+
+TimeControl::TimeControl()
+  : EPSILON(1.0e-10),
+    reset_counter(true),
+    is_active(false),
+    start_time(std::numeric_limits<double>::max()),
+    trigger_every_time_interval(-1.0),
+    trigger_every_time_steps(std::numeric_limits<unsigned int>::max()),
+    counter(0),
+    last_timestep_number_to_evaluate(dealii::numbers::invalid_unsigned_int)
+{
+}
+
+void
+TimeControl::reinit(bool const         is_active_,
+                    double const       start_time_,
+                    double const       trigger_every_time_interval_,
+                    unsigned int const trigger_every_time_steps_)
+{
+  is_active = is_active_;
+
+  if(is_active_)
+  {
+    if(trigger_every_time_steps_ == std::numeric_limits<unsigned int>::max() ||
+       trigger_every_time_steps_ == 0)
+    {
+      AssertThrow(trigger_every_time_interval_ > 0.0 &&
+                    trigger_every_time_interval_ < std::numeric_limits<double>::max(),
```

Comment:
I seem not to be able to mark more lines. What I am referring to with the questions are the two asserts in the reinit function

Reply:
I think these checks mean that the user sets exactly on option (time steps or time interval) in case that the postprocessing routine is active. So we also need it for the unsteady case. However, it does not really make sense in the steady case. We just want to activate a postprocessing routine without specifying the number of time steps or a time interval.

Reply:
So the question is: Should the user care about setting the time_interval or time_step option such that the asserts are fulfilled also in the steady case. Is this currently done?

Reply:
The asserts should not be checked for steady problems in my opinion. It is probably difficult to pass the information steady/unsteady to the `reinit()` function. It would be simpler to include the checks in the `needs_evaluation()` function.

Reply:
[f6b2eae](https://github.com/exadg/exadg/pull/262/commits/f6b2eae4cde8451c53c0d307d844518d4d702abc)